websites/Z/Zee5/presence.ts

```diff
@@ -0,0 +1,103 @@
+const presence = new Presence({
+		clientId: "1103408220921397349",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+/**
+ * Get Timestamps
+ * @param {Number} videoTime Current video time seconds
+ * @param {Number} videoDuration Video duration seconds
+ */
+function getTimestamps(videoTime: number, videoDuration: number): number[] {
+	const startTime = Date.now();
+	return [
+		Math.floor(startTime / 1000),
+		Math.floor(startTime / 1000) - videoTime + videoDuration,
+	];
+}
+
+const getVideoStatus = (
+	presenceData: PresenceData,
+	video: HTMLVideoElement
+) => {
+	if (video.paused) {
+		presenceData.smallImageKey = "paused";
+		presenceData.smallImageText = "paused";
+		delete presenceData.startTimestamp;
+	} else {
+		const [startTimestamp, endTimestamp] = getTimestamps(
+			Math.floor(video.currentTime),
+			Math.floor(video.duration)
+		);
+		presenceData.startTimestamp = startTimestamp;
+		presenceData.endTimestamp = endTimestamp;
+		presenceData.smallImageKey = "playing";
+		presenceData.smallImageText = "playing";
+	}
+	return presenceData;
+};
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/H9fXid0.png",
+		startTimestamp: browsingTimestamp,
+	};
+
+	if (document.location.pathname === "/")
```

Comment:
As `document.location` is used many times in this presence, destructure the properties for simplicity/consistency.
```suggestion
	const { pathname } = document.location;
	if (pathname === "/")
```

Reply:
Also do this for all other uses of pathname.

Reply:
Also do this for all other uses of pathname.
so
```ts
// all these:
document.location.pathname
// change to:
pathname
```